test/neg-http.mjs

```diff
@@ -124,102 +138,106 @@ describe('Testing the challenge http API', () => {
-            console.log(`Created flawed block with invalid historic root ${res.block.blockHash}`);
-          } else if (counter === 6) {
-            res = await createBadBlock('IncorrectPublicInputHash', block, transactions);
-            topicsBlockHashIncorrectPublicInputHash = res.block.blockHash;
-            txDataToSign = res.txDataToSign;
-            console.log(
-              `Created flawed block with incorrect public input hash and blockHash ${res.block.blockHash}`,
-            );
-          } else if (counter === 7) {
-            res = await createBadBlock('IncorrectProof', block, transactions, {
-              proof: duplicateTransaction.proof,
-            });
-            topicsBlockHashIncorrectProof = res.block.blockHash;
-            txDataToSign = res.txDataToSign;
-            console.log(
-              `Created flawed block with incorrect proof and blockHash ${res.block.blockHash}`,
-            );
-          } else if (counter === 8) {
-            res = await createBadBlock('DuplicateNullifier', block, transactions, {
-              duplicateNullifier,
-            });
-            topicsBlockHashDuplicateNullifier = res.block.blockHash;
-            txDataToSign = res.txDataToSign;
-            console.log(
-              `Created flawed block with duplicate nullifier and blockHash ${res.block.blockHash}`,
-            );
-          } else if (counter === 9) {
-            res = await createBadBlock('IncorrectLeafCount', block, transactions);
-            topicsBlockHashIncorrectLeafCount = res.block.blockHash;
-            txDataToSign = res.txDataToSign;
-            console.log(
-              `Created flawed block with incorrect leaf count and blockHash ${res.block.blockHash}`,
-            );
-          } else {
-            txDataToSign = msg.txDataToSign;
-            console.log(`Created good block with blockHash ${block.blockHash}`);
-          }
-          await submitTransaction(txDataToSign, privateKey, stateAddress, gas, BLOCK_STAKE);
-          counter++;
-          // console.log('tx hash of propose block is', txReceipt.transactionHash);
-        } else if (type === 'commit') {
-          await submitTransaction(txDataToSign, privateKey1, challengeAddress, gas);
-        } else if (type === 'challenge') {
-          await submitTransaction(txDataToSign, privateKey1, challengeAddress, gas);
-          // When a challenge succeeds, the challenger is removed. We are adding them back for subsequent for challenges
-          const result = await chai
-            .request(optimistUrl)
-            .post('/proposer/register')
-            .send({ address: myAddress });
-          txToSign = result.body.txDataToSign;
-          await submitTransaction(txToSign, privateKey, proposersAddress, gas, bond);
-          // console.log('tx hash of challenge block is', txReceipt.transactionHash);
-        } else throw new Error(`Unhandled transaction type: ${type}`);
+        try {
```

Comment:
Why do we need `try catch` around the websocket code?

Reply:
whenever error occurs in websocket 'onmessage' emitHandler it does not terminates test suite( throw does not propagate to integration test execution). instead next dependent test check fails.
I added try - catch with console.log to easy debug at such cases